base/libgit2/strarray.jl

```diff
@@ -2,7 +2,11 @@
 
 function StrArrayStruct{T<:AbstractString}(strs::T...)
     strcount = length(strs)
-    map(s->Base.unsafe_convert(Cstring, String(s)), strs) # check for null-strings
+    for s in strs
+        if Base.containsnul(s)
+            throw("embedded NULs are not allowed in C strings: $(repr(s))")
+        end
+    end
```

Comment:
You are right. I'm pretty sure `libgit2` is full of these though..... a few type declarations make it almost impossible to implement the right rooting.